src/lib/reducers/maxes.reducer.ts

```diff
@@ -0,0 +1,155 @@
+import { KeyValuePair } from "@react-native-async-storage/async-storage/lib/typescript/types";
+
+export enum MaxesViewReducerActions {
+  SET_LIFT,
+  SET_WEIGHT,
+  SET_STORAGE,
+  SET_FAILED_LOAD,
+  SET_ADDING_STATE
+}
+
+const {
+  SET_LIFT,
+  SET_WEIGHT,
+  SET_STORAGE,
+  SET_FAILED_LOAD,
+  SET_ADDING_STATE,
+} = MaxesViewReducerActions;
+
+interface MaxesViewState {
+  lift: string
+  weight: string
+  dataInStorage: KeyValuePair[] | []
+  didLoadingFail: boolean
+  isStorageAdding: boolean
+}
+
+interface MaxesViewAction {
+  type: MaxesViewReducerActions,
+  payload: any,
+}
+
+export const maxesReducer = (
+  state: MaxesViewState, action: MaxesViewAction
+) => {
+
+  const { type, payload } = action;
+
+  // destructure 'state' prop here if need to
+  // get "old" (i.e. the current values in the component's state object).
+  // const { x, y, ... } = state;
+
+
+  const {
+    newLiftName, newWeightAmount, 
+    newAddingState, newFailState, newStorageData
+  } = payload;
+  
+  // To help log the reducer's behavior in "dev". 
+  // We can set env variable to toggle this off in "prod".
+  const returnPreviousState = (stateProp: string, errorMessage?: string)  => {
+    console.log(`> MaxesReducer: Failed to update state for: ${stateProp}`);
+    if (errorMessage) console.log(`> ${errorMessage}`);
+    console.log('> Returning previous state with no changes.');
+    return { ...state };
+  }
+
+
+  // Actions the reduces can take to update the current
+  // state object tied to the component the reducer is
+  // attached to.
+  switch (type) {
+    
+    case SET_LIFT: {
+
+      if (!newLiftName) {
+        return returnPreviousState(
+          'lift', 'No new state value for -lift- was provided'
+        );
+      }
+
+      const regex = /^.{3,60}$/; // Lift name - btwn 3 & 60 characters 
+      const isValidLiftName = regex.test(newLiftName);
```

Comment:
This generally makes it so you can't add a new lift. The condition when you CAN is when you spam keys super fast. This then makes it so you can't delete the first 3 chars of your new lift...